File: ga/gpt3_5_result_2024_6/result/1968C_cpp_gpt/46_1968C.cpp
use std::io::{self, BufRead};

const N: usize = 500;
const X: usize = 500;
const A: i64 = 1000000000;

fn proc() {
    let mut x = vec![0; N];
    let mut a = vec![0; N];

    let mut line = String::new();
    io::stdin().lock().read_line(&mut line).unwrap();
    let n: usize = line.trim().parse().unwrap();

    line.clear();
    io::stdin().lock().read_line(&mut line).unwrap();
    let mut input_iter = line.split_whitespace().skip(1);

    for i in 1..n {
        x[i] = input_iter.next().unwrap().parse().unwrap();
    }

    a[0] = x[1] + 1;

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i] as usize) || a[i] != a[i - 1] as usize {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    let t: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}

File: ga/4omini_result_2024_7/result/1968C_cpp_mini/46_1968C.rs
use std::io;

const N: usize = 500;
const A: usize = 1_000_000_000;

fn proc() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut x = vec![0; N];
    let mut a = vec![0; N];

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let x_values: Vec<i32> = input.trim().split_whitespace()
                                    .map(|s| s.parse().unwrap())
                                    .collect();
    
    for i in 0..n - 1 {
        x[i + 1] = x_values[i]; // Adjust index to match x[1] to x[n-1]
    }
    
    a[0] = x[1] + 1; // The first element a[0]
    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j as i32 * a[i - 1];
            if (i + 1 < n) && x[i + 1] < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!(); // Print newline at the end
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}